windows/binary_security_descriptor.ksy

```diff
@@ -0,0 +1,244 @@
+    size: 8
+  - id: len_binarysd
+    type: u4
+  - id: padding
+    type: u8
+  - id: binarysd
+    type: binarysd
+    size: len_binarysd
+types:
+  binarysd:
+    seq:
+    - id: revision
+      type: u1
+    - id: reserved
+      size: 1
+      contents: [0]
+    - id: flags
+      type: binarysd_flags
+    - id: owner_offset
+      type: u4
+    - id: group_offset
+      type: u4
+    - id: sacl_offset
+      type: u4
+    - id: dacl_offset
+      type: u4
+    instances:
+      owner:
+        pos: owner_offset
+        type: sid
+        if: 'owner_offset > 0'
+      group:
+        pos: group_offset
+        type: sid
+        if: 'group_offset > 0'
+      dacl:
+        pos: dacl_offset
+        type: acl
+        if: 'dacl_offset > 0'
+      sacl:
+        pos: sacl_offset
+        type: acl
+        if: 'sacl_offset > 0'
+  binarysd_flags:
+    seq:
+      - id: dacl_trusted
+        type: b1
+        doc: DACL Trusted
+      - id: server_security
+        type: b1
+        doc: Server Security
+      - id: sacl_defaulted
+        type: b1
+        doc: SACL defaulted
+      - id: sacl_present
+        type: b1
+        doc: SACL present
+      - id: dacl_defaulted
+        type: b1
+        doc: DACL defaulted
+      - id: dacl_present
+        type: b1
+        doc: DACL present
+      - id: group_defaulted
+        type: b1
+        doc: Group defaulted
+      - id: owner_defaulted
+        type: b1
+        doc: Owner defaulted
+      - id: self_relative
+        type: b1
+        doc: Self-Relative
+      - id: control_valid
+        type: b1
+        doc: Control Valid
+      - id: sacl_protected
+        type: b1
+        doc: SACL-protected
+      - id: dacl_protected
+        type: b1
+        doc: DACL-protected
+      - id: sacl_auto_inherited
+        type: b1
+        doc: SACL auto-inherited
+      - id: dacl_auto_inherited
+        type: b1
+        doc: DACL auto-inherited
+      - id: sacl_inheritance_required
+        type: b1
+        doc: SACL Inheritance Required
+      - id: dacl_inheritance_required
+        type: b1
+        doc: DACL Inheritance Required
+  acl:
+    seq:
+      - id: revision
+        type: u1
+      - id: reserved
+        size: 1
+        contents: [0]
+      - id: acl_size
+        type: u2
+      - id: ace_count
+        type: u2
+      - id: reserved2
+        size: 2
+        contents: [0, 0]
+      - id: ace_list
+        type: ace_list
+        size: acl_size - 8
+  ace_list:
+    seq:
+      - id: ace
+        type: ace
+        repeat: eos
+  ace:
+    doc-ref: https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-dtyp/628ebb1d-c509-4ea0-a10f-77ef97ca4586
+    seq:
+      - id: access_allow_type
+        type: u1
+      - id: flags
+        type: ace_flags
+      - id: ace_size
+        type: u2
+      - id: access_mask
+        type: access_mask
+      - id: sid
+        type: sid
+        size: ace_size - 8
+  ace_flags:
+    seq:
+        - id: failed_access
```

Comment:
Should binary fields have james like questions?

Reply:
I took the names from the flag names in the Microsoft documentation